### PR TITLE
Cleaned up tab add and delete UX

### PR DESCRIPTION
**Pull request by shueja-personal**

### CODE REVIEW

File: packages/dashboard/src/root/navbar.js [hunk 1]
```diff
@@ -166,7 +211,11 @@ class DashboardNavbar extends LitElement {
         <div class="tabs">
           <vaadin-tabs .selected=${this.selectedTabIndex} @selected-changed=${this.#onTabChange}>
             ${tabs.map(({ label }, index) => html`
-            <vaadin-tab value=${index}>${label}</vaadin-tab>
+            <vaadin-tab value=${index}>${label} 
+              <vaadin-button class="delete-tab-button" theme="icon tertiary" aria-label="Delete Tab" deletes=${label} @click=${()=>this.deleteTab(label)}
+              ?disabled=${!this.dashboard.isDrawerOpened()}>
+              <vaadin-icon icon="vaadin:close"></vaadin-icon>
+            </vaadin-tab>
```
Comment (amorygalili) [inline packages/dashboard/src/root/navbar.js:218]: This looks good. Could the icon be made a little smaller? I think it looks better when it's less prominent:

tabs-with-x

Could you also add `cursor: pointer` to the icon styles?

File: packages/dashboard/src/root/navbar.js [hunk 2]
```diff
@@ -166,7 +211,11 @@ class DashboardNavbar extends LitElement {
         <div class="tabs">
           <vaadin-tabs .selected=${this.selectedTabIndex} @selected-changed=${this.#onTabChange}>
             ${tabs.map(({ label }, index) => html`
-            <vaadin-tab value=${index}>${label}</vaadin-tab>
+            <vaadin-tab value=${index}>${label} 
+              <vaadin-button class="delete-tab-button" theme="icon tertiary" aria-label="Delete Tab" deletes=${label} @click=${()=>this.deleteTab(label)}
```
Comment (amorygalili) [inline packages/dashboard/src/root/navbar.js:215]: I feel like the name logic in `onAddTab` and `deleteTab` function is overkill, and `() =>  removeElement(tabs[index], connector)` would probably work fine here. What do you think?

Comment (shueja-personal) [reply packages/dashboard/src/root/navbar.js:215]: It’s overkill if the ordering in #getTabs is guaranteed to be equal to the indices in the html. You’re referring to the search by name?

**Comment by shueja-personal** [comment]
Rebased, tested and working with the snapping code

**Pull request merged**